### ISSUE TITLE
No link to the source code anymore

### ISSUE THREAD

**Issue by couet** [doxygen/doxygen#8679]
I have installed the master version of doxygen (1.9.2) to take advantage of a [bug fix not yet released](https://github.com/doxygen/doxygen/issues/8620). 

With that new version I saw that the C++ methods having only a `\brief` as documentation did not have a detailed description anymore. Adding `ALWAYS_DETAILED_SEC` fixed this problem.

But there is still an issue for methods having no help at all, not even a `\brief` . Such methods do not have a detailed description anymore. One can say it's normal as there is no doc why having a detailed description ?

Indeed the detailed description is needed in a such case because it is the only way to access the source code. Seeing the source code for such methods can be very important to understand the algorithm or the logic used. Specially in a Physics Data analysis system like [ROOT](https://root.cern). Such methods do not need a `\brief` or a help because their names are explicit enough and a `\brief` will be a copy of the method name.

So, we do not need the detailed description for such methods, but we need a direct link to the code. 
Is there a way to do it ?

**Comment by albert-github** [comment]
- Which version of 1.9.2 did you install (i.e. complete output of `doxygen -v` is relevant)?
- You probably have `EXTRACT_ALL` set, but the relevant functions don't have a detailed description, so no need for a detailed section. The `EXTRACT_ALL`, incorrectly, triggered a detailed section in the past.

Without an example it is hard to tell. So:
- Can you please attach a, small, self contained example (source+configuration file in a tar or zip) that allows us to reproduce the problem? Please don't add external links as they might not be persistent.

**Comment by couet** [comment]
Thanks for your fast reply. Sorry to not have been precise enough. Let me complete my post.

  - `doxygen -v` gives me:

```
% doxygen -v                    
1.9.2 (97dffa0977cc41729942cf3a8c87232eb7edc09d)
```

   - Yes I have `EXTRACT_ALL` set.

  - A small reproducer is:
```
/// \class A
/// Class A
class A {
public:
   /// \brief method 1 in A
   void method1(int i){}
   void method2(int i){}
};
```

  it gives the following output:
<img width="953" alt="Screenshot 2021-07-21 at 13 00 12" src="https://user-images.githubusercontent.com/4697738/126478665-289ee899-845d-4426-baf8-c5d17e291206.png">

As can you see: 
   - `method1` has a `\ brief` and we can access to the code, 
   - `method2` has no `\brief` we can see it in the doc but we cannot access the code.

**Comment by albert-github** [comment]
I assume you have at least the following settings in your doxygen configuration file:
```
SOURCE_BROWSER=YES
ALWAYS_DETAILED_SEC=YES
```
as  with those settings I can reproduce your problem.
(for this reason I, nearly, always request: "Can you please attach a, small, self contained example (source+configuration file in a tar or zip) that allows us to reproduce the problem? Please don't add external links as they might not be persistent." so we are talking about the same input).

I remember that there was a problem with the `SOURCE_BROWSER`  reference output. For HTML only `SOURCE_BROWSER` has to be set to get the link, but for e.g. besides the `SOURCE_BROWSER` also the `LATEX_SOURCE_CODE` has to be set (similar for e.g. RTF, DOCBOOK, XML) and the routine detecting whether or detailed description should be present does not know about the type of output that is requested
The ALWAYS_DETAILED_SEC` looks like to be the solution, but this is only triggered when there is brief documentation present:
```
         (!briefDescription().isEmpty() &&           // has brief docs
          (alwaysDetailedSec &&                      // they are visible in
           (repeatBrief ||                           // detailed section or
            !briefMemberDesc                         // they are explicitly not
           )                                         // shown in brief section
          )
         ) 
```

It looks like there is another problem as well, in case of e.g. LaTeX the source code is only generated in case `EXTRACT_ALL` is set as well or a brief documentation is present for the file.

So all in all some optimization potential here.

**Comment by couet** [comment]
Yes I have:
```
SOURCE_BROWSER        = YES
ALWAYS_DETAILED_SEC = YES
```

Does that mean there is no workaround I can use now to fix this issue ? 
Should I wait for a further fix ?

One thing which coud be done is:  if `SOURCE_BROWSER`  and `EXTRACT_ALL` are `ON` then, for the methods which do not have any documentation, the method name can be a link which point directly to the source code as, for such methods, the detailed description does not make sense.

**Comment by albert-github** [comment]
There is no workaround other than manipulating the code (i.e. in `memberdef.cpp` in the function `MemberDefImpl::isDetailedSectionLinkable` adding a test for `SOURCE_BROWSER` or `EXTRACT_ALL` in the if statement. probably best here is `SOURCE_BROWSER` at least when you use only HTML).

**Comment by couet** [comment]
Ok. Thanks for your help and suggestions. I will look at it and possibly propose a PR if I managed to fix it.

**Comment by albert-github** [comment]
Currently I'm looking into the problem regarding "It looks like there is another problem as well, in case of e.g. LaTeX the source code is only generated in case EXTRACT_ALL is set as well or a brief documentation is present for the file.", when this is fixed I can look a step further.

**Comment by albert-github** [comment]
As required preliminary, but required,work for the problem in this issue (so links can be visible in all output formats)  I've just pushed a proposed patch, pull request #8683

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.2.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).

**Comment by couet** [comment]
I tested. It is fixed. Thanks
